onnxscript/rewriter/generic_pattern.py

```diff
@@ -294,6 +294,78 @@ def make_node(
         return proto
 
 
+class PatternMatchResult:
+    """Stores information about a match if a match was successful."""
+
+    def __init__(
+        self,
+        pattern: "GenericPattern",
+        model_nodes: typing.Sequence[oir.Node],
+        pattern_nodes: typing.Sequence[oir.Node],
+        pattern_input_names: typing.Sequence[str],
+        pattern_output_names: typing.Sequence[str],
+    ):
+        assert len(model_nodes) == len(pattern_nodes), (
+            f"The number of matched nodes in the model {len(model_nodes)} "
+            f"should the same as the number of nodes in the pattern "
+            f"{len(pattern_nodes)}."
+        )
+        self.pattern = pattern
+        self.model_nodes = model_nodes
+        self.pattern_nodes = pattern_nodes
+        self.pattern_input_names = pattern_input_names
+        self.pattern_output_names = pattern_output_names
+        self.kwargs = {}
+
+        matched_pattern_to_model_name: dict[str, str] = {}
+        for gn, pn in zip(model_nodes, pattern_nodes):
+            assert (
+                gn.op_type == pn.op_type
+            ), f"Unexpected type mismatch {gn.op_type!r} != {pn.op_type!r}"
```

Comment:
nit: Avoid abbreviations

Reply:
nit: if abbreviations are to be used, would be helpful to change `gn` to `mn` ... to match the first letter of `model_nodes` ... or change `model_nodes` to `graph_nodes` (which is where the `g` is coming from, I guess).